### ISSUE TITLE
Load route templateUrl after the route resolve it's done.

### ISSUE THREAD

**Issue by cssensei** [angular/angular.js#11179]
When you would like to modify the `route.templateUrl` based on some API data that will be evaluated in the `route.resolve` (like user access level)  the html template it's already loaded with `$templateCache` generating a `$route.current.loadedTemplate`. We should be able to modify that value, or at least have a `reload` method just for the `templateUrl`.

```
app.config(['$routeProvider',
	function($routeProvider) {
		$routeProvider.
		when('/someRoute', {
			templateUrl: 'views/legacyView.html', //This is my default (legacy view)
			controller: 'someRouteCtrl',
			resolve: ['UserSession', '$route', '$q',
				function(UserSession, $route, $q) {
					var routeDone = $q.defer();
					UserSession.getAccessLevel().then(function(accessLevel) {
						if (accessLevel == 'CanViewFacelift') {
							//This doesn't affect the view because
							//views/legacyView.html it's already in DOM
							$route.current.templateUrl = 'views/faceliftView.html';
						}
						routeDone.resolve(accessLevel);
					}, function() {
						routeDone.reject('No access!');
					})
					return routeDone.promise;
				}
			]
		});
	}
]);
```

**Comment by Narretz** [comment]
This could potentially impact perceived responsiveness for the user, because the template would load after the resolves and not in parallel to them. 
However, apps that use a prefilled templateCache wouldn't have this problem.

**Comment by Narretz** [comment]
Duplicate of #9091
